GVFS.ruleset

```diff
@@ -0,0 +1,102 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<RuleSet Name="Rules for GVFS" Description="Code analysis rules for GVFS projects" ToolsVersion="15.0">

```

Comment:
Are these rules a conversion from the old Settings.Stylecop list or a list of default/recommended warnings? If it's a list of defaults (which I suspect due to the number of rules here vs what we have today), can we start by turning on only the rules we had on in legacy Stylecop and make the infrastructure changes first then as a follow up, have a discussion on what new rules we should turn on and make the codebase clean for those rules?

Reply:
The `CA*` rules are from the "Managed Binary Analysis" rules that are in the `Microsoft Managed Minimum / Recommended Rules" ruleset. I am still looking at how these rules interact with our build - the MSDN documentation seems to recommend that some of these should be included in custom rulesets.

The stylecop rules have a default severity. For most rules the default severity is "warning". The ones I have overriden to  "None" are those that we are not clean on currently. For some projects, I did clean up some of the existing violations, especially for rules that I thought we generally enforced already.